spec/rspec/expectations/handler_spec.rb

```diff
@@ -130,6 +130,21 @@ module Expectations
 
           RSpec::Expectations::PositiveExpectationHandler.handle_matcher(actual, matcher, failure_message)
         end
+
+        it "add custom failure message to mock matcher if there is one" do
+          error_generator = instance_double("RSpec::Mocks::ErrorGenerator", :opts => { :message => {} })
+          verifying_message_expectation = instance_double(
+            "RSpec::Mocks::VerifyingMessageExpectation",
+            :error_generator => error_generator
+          )
+          matcher = double("matcher", :failure_message => "message", :matches? => verifying_message_expectation)
+
+          actual = Object.new
+
+          handle_matcher_answer = RSpec::Expectations::PositiveExpectationHandler.handle_matcher(actual, matcher, "custom")
```

Comment:
I'd keep this name in line with above:
```suggestion
          handle_matcher_result = RSpec::Expectations::PositiveExpectationHandler.handle_matcher(actual, matcher, "custom")
```